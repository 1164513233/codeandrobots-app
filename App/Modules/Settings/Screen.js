import React, { Component } from 'react'
import { View, Text } from 'react-native'
import PropTypes from 'prop-types'

import { appInfo } from 'App/Services/Properties'

import { Container, Footer, CompactListItem, Links, Link } from 'App/Components'

import s from './Styles'

export default class Screen extends Component {
  static propTypes = {
    onChooseRobotPress: PropTypes.func.isRequired,
    onShareAppPress: PropTypes.func.isRequired,
    onFeedbackPress: PropTypes.func.isRequired,
    onFAQPress: PropTypes.func.isRequired,
    onSupportPress: PropTypes.func.isRequired,
    onAboutPress: PropTypes.func.isRequired,
    onPrivacyPress: PropTypes.func.isRequired
  }

  render () {
    const {
      onChooseRobotPress,
      onShareAppPress,
      onFeedbackPress,
      onFAQPress,
      onSupportPress,
      onAboutPress,
      onPrivacyPress} = this.props
    return (
      <Container>
        <CompactListItem title='Choose the robot you want to play with' onPress={onChooseRobotPress} />
        <CompactListItem title='Share app with a friend' onPress={onShareAppPress} />
        <CompactListItem title='Share your ideas with us' onPress={onFeedbackPress} />
        <View style={s.appVersionView}>
          <Text style={s.appVersion}>{`App Version ${appInfo()}`}</Text>
        </View>
        <Footer>
          <Links>
            <Link text='FAQ' onPress={onFAQPress} />
            <Link text='Support' onPress={onSupportPress} />
            <Link text='About' onPress={onAboutPress} />
            <Link text='Privacy' onPress={onPrivacyPress} />
          </Links>
        </Footer>
      </Container>
    )
  }
}
