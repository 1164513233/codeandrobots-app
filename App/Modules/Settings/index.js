export { default as Settings } from './Screens/Containers'
