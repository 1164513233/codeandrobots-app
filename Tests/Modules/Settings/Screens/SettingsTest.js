import React from 'react'

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Settings from 'App/Modules/Settings/Screens/Components'

describe('Settings', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <Settings
        onShareAppPress={() => {}}
        onFeedbackPress={() => {}}
        onFAQPress={() => {}}
        onSupportPress={() => {}}
        onAboutPress={() => {}} />)
  })

  it('Component renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
