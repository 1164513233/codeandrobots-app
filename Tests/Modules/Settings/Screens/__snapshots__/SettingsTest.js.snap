// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Component renders correctly 1`] = `
<Container>
  <Setting
    onPress={[Function]}
    text="Share app with a friend"
  />
  <Setting
    onPress={[Function]}
    text="Share your ideas with us"
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 224,
        "display": "flex",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#9B9B9B",
          "fontFamily": "Roboto-Medium",
          "fontSize": 14,
        }
      }
    >
      App Version N/A (N/A)
    </Text>
  </View>
  <Footer>
    <Links>
      <Link
        onPress={[Function]}
        text="FAQ"
      />
      <Link
        onPress={[Function]}
        text="Support"
      />
      <Link
        onPress={[Function]}
        text="About"
      />
    </Links>
  </Footer>
</Container>
`;
